httpclient/request.go

```diff
@@ -33,6 +33,17 @@ func WithRequestHeaders(headers map[string]string) DoOption {
 	})
 }
 
+// WithToken uses the specified golang.org/x/oauth2 token on a request
+func WithToken(token *oauth2.Token) DoOption {
+	visitor := WithRequestVisitor(func(r *http.Request) error {
+		auth := fmt.Sprintf("%s %s", token.TokenType, token.AccessToken)
+		r.Header.Set("Authorization", auth)
+		return nil
+	})
+	visitor.isAuthOption = true
+	return visitor
+}
```

Comment:
Alternatively, we could do the following:

```go
func WithToken(token *oauth2.Token) DoOption {
  return WithTokenSource(oauth2.StaticTokenSource(token))
}
```

Or maybe completely remove the function and use `WithTokenSource(oauth2.StaticTokenSource(token))` directly instead of `WithToken(token)`?

Reply:
If you look at it closely, `WithTokenSource` does not set `visitor.isAuthOption = true`.
This is because `IsAuthOption` is a new flag. If should be added there, but it requires delicate testing.

But we can still simplify it like this: 

```
func WithToken(token *oauth2.Token) DoOption {
  visitor := WithTokenSource(oauth2.StaticTokenSource(token))
  visitor.isAuthOption = true
  return visitor
}
```

Reply:
Oh I missed that, thanks for pointing it out!